src/core/Expectation.js

```diff
@@ -62,8 +62,12 @@ getJasmineRequireObj().Expectation = function(j$) {
       );
     }
 
-    if (!j$.isPromiseLike(this.expector.actual)) {
```

Comment:
I hear that you want to be able to support matchers that do async work themselves and not just matchers that work on actuals that are promises. I'm not necessarily opposed to that, and I think there's a case to be made for doing that at the same time as adding custom async matchers if we're going to do it at all. But as it stands right now, this PR makes the error message less clear if any of the existing async matchers are called with an actual that isn't a promise.

Before:
```
Error: Expected expectAsync to be called with a promise.
    at <Jasmine>
    at UserContext.<anonymous> (http://localhost:8888/__spec__/core/integration/EnvSpec.js:2499:12)
    at <Jasmine>
```

After:
```
TypeError: actual.then is not a function
    at <Jasmine>
    at UserContext.<anonymous> (http://localhost:8888/__spec__/core/integration/EnvSpec.js:2614:28)
    at <Jasmine>
```

So I think that what we should do for now is either put the type guard back, or add a similar check to each of the existing async matchers. Either way, when somebody does something like `expectAsync('not a promise').toBeResolved()`, the error message should clearly indicate that the actual needs to be a promise.